tests/pytests/test_mp2.py

```diff
@@ -1,327 +1,641 @@
-    pytest.param({'driver': 'gradient', 'subject': 'hf',     'options': {'reference': 'rhf',  'mp2_type': 'df',   'qc_module': 'dfmp2', 'freeze_core': 'true',  'scf_type': 'df'}}, id='mp2 grad  rhf df    fc: * dfmp2'),
-    pytest.param({'driver': 'gradient', 'subject': 'hf',     'options': {'reference': 'rhf',  'mp2_type': 'df',   'qc_module': 'occ',   'freeze_core': 'false', 'scf_type': 'df'}}, id='mp2 grad  rhf df   nfc:   occ'),
-    pytest.param({'driver': 'gradient', 'subject': 'hf',     'options': {'reference': 'rhf',  'mp2_type': 'df',   'qc_module': 'dfmp2', 'freeze_core': 'false', 'scf_type': 'df'}}, id='mp2 grad  rhf df   nfc: * dfmp2'),
-    pytest.param({'driver': 'gradient', 'subject': 'bh_h2p', 'options': {'reference': 'uhf',  'mp2_type': 'conv', 'qc_module': 'occ',   'freeze_core': 'false', 'scf_type': 'pk'}}, id='mp2 grad  uhf conv nfc: * occ'),
-    pytest.param({'driver': 'gradient', 'subject': 'bh_h2p', 'options': {'reference': 'uhf',  'mp2_type': 'df',   'qc_module': 'occ',   'freeze_core': 'true',  'scf_type': 'df'}}, id='mp2 grad  uhf df    fc: * occ'),
-    pytest.param({'driver': 'gradient', 'subject': 'bh_h2p', 'options': {'reference': 'uhf',  'mp2_type': 'df',   'qc_module': 'occ',   'freeze_core': 'false', 'scf_type': 'df'}}, id='mp2 grad  uhf df   nfc: * occ'),
-    # For gradients, this method would be found in the procedures table but would return a ManagedMethodError from proc.py. Normally, this would confuse the analytic-or-findif logic in gradient().
-    #   This scenario is now managed, and this test case makes sure its routing stays managed.
-    pytest.param({'driver': 'gradient', 'subject': 'bh_h2p', 'options': {'reference': 'rohf',  'mp2_type': 'df',   'qc_module': 'occ',   'freeze_core': 'false', 'scf_type': 'df', 'points': 5}}, id='mp2 grad rohf df   nfc: findif'),
-])  # yapf: disable
-def test_mp2_module(inp, clsd_open_pmols, request):
+_nyi1 = "fc conv mp2 gradients NYI"
+_nyi2 = "rohf mp2 gradients NYI"
+_nyi3 = "cd mp2 gradients NYI"
+_nyi4 = "spin components rhf mp2 energies NYI"
+
+
+@pytest.mark.parametrize("dertype", [0,], ids=["ene0"])
+@pytest.mark.parametrize(
+    "inp",
+    [
+        # yapf: disable
+        ######## Are scf_types managed properly by proc.py?
+
+        pytest.param({"driver": "energy", "subject": "hf",   "options": {"reference": "rhf",  "mp2_type": "df",   "qc_module": "dfmp2", "freeze_core": "false",  "scf_type": "pk",     },             }, id="mp2  rhf   pk/df   rr dfmp2",),
+        pytest.param({"driver": "energy", "subject": "hf",   "options": {"reference": "rhf",  "mp2_type": "df",   "qc_module": "dfmp2", "freeze_core": "false",  "scf_type": "direct", },             }, id="mp2  rhf drct/df   rr dfmp2",),
+        pytest.param({"driver": "energy", "subject": "hf",   "options": {"reference": "rhf",  "mp2_type": "df",   "qc_module": "dfmp2", "freeze_core": "false",  "scf_type": "df",     },             }, id="mp2  rhf   df/df   rr dfmp2",),
```

Comment:
Is it possible to use a loop in here, instead of explicit `pytest.param` for each combination?

Reply:
this scf_type test is probably regular enough to run in an extra `pytest.param` loop, but many aren't with the error/wrong exceptions, so I'd like to keep the outer forgettable loops at mol/basis/dertype, I think.